src/Microsoft.TestPlatform.Client/DesignMode/DesignModeTestEventsRegistrar.cs

```diff
@@ -4,22 +4,21 @@
 namespace Microsoft.VisualStudio.TestPlatform.Client.DesignMode
 {
     using Microsoft.VisualStudio.TestPlatform.Common.Interfaces;
+    using Microsoft.VisualStudio.TestPlatform.CommunicationUtilities;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
     using Microsoft.VisualStudio.TestPlatform.ObjectModel.Logging;
 
-    /// <summary>
-    /// Registers the discovery and test run events for design mode flow
-    /// </summary>
-    public class DesignModeTestEventsRegistrar : ITestDiscoveryEventsRegistrar, ITestRunEventsRegistrar
+    internal class IdentifiableDesignModeTestEventsRegistrar : ITestDiscoveryEventsRegistrar, ITestRunEventsRegistrar
     {
+        private string testRunId;
         private IDesignModeClient designModeClient;
 
-        public DesignModeTestEventsRegistrar(IDesignModeClient designModeClient)
+        public IdentifiableDesignModeTestEventsRegistrar(IDesignModeClient designModeClient, string testRunId)
```

Comment:
I'd keep `recipient`  for clarity or do you have a reason to change the name?

Reply:
I thought I need to rename this type, but I don't need to do that, so I will rename that back. And the testRunId was a previous design. So I will change that as well.